insights/combiners/package_provides_java.py

```diff
@@ -0,0 +1,48 @@
+"""
+Combiner for packages which provide java
+========================================
+
+Combiner for collecting all the PackageProvidesJava objects with two properties, one is running java command
+and the other is package name that provides corresponding java command.
+It provides an interface to get all the valid such kind of objects.
+
+Examples:
+    >>> PACKAGE_COMMAND_MATCH_1 = '''/usr/lib/jvm/jre/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'''
+    >>> PACKAGE_COMMAND_MATCH_2 = '''/usr/lib/jvm/java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64/bin/java java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'''
+    >>> pack1 = PackageProvidesJava(context_wrap(PACKAGE_COMMAND_MATCH_1))
+    >>> pack2 = PackageProvidesJava(context_wrap(PACKAGE_COMMAND_MATCH_2))
+    >>> shared = [{PackageProvidesJavaAll: [pack1, pack2]}]
+    >>> packages = shared[PackageProvidesJavaAll]
+    >>> package = packages[0]
+    >>> package.command
+    '/usr/lib/jvm/jre/bin/java'
+    >>> package.package
+    'java-1.8.0-openjdk-headless-1.8.0.141-3.b16.el6_9.x86_64'
+"""
+
+from insights.core.plugins import combiner
+from insights.parsers.package_provides_java import PackageProvidesJava
+
+
+@combiner(PackageProvidesJava)
+class PackageProvidesJavaAll(object):
+    """
+    Combiner for collecting all the PackageProvidesJava objects with two properties, one is running java command
+    and the other is package name that provides corresponding java command.
+
+    Attributes:
+        data (list): List of objects containing package name and its providing java command
+    """
+
+    def __init__(self, package_provides_java):
+        packages_list = []
+        for package in package_provides_java:
+            packages_list.append(package)
+        self.data = packages_list
+
+    def __getitem__(self, index):
```

Comment:
How about adding an interface like `get_package(command)`, then you can call it directly after getting the command from `PS`, e.g.: get_package(ps_obj.row["COMMAND"])

Reply:
And you can mixin it with LegacyItemAccess to save some duplicated codes.

Reply:
@xiangce Good suggestion. I add a get_package() interface and mixin LefacyItemAccess as well.